plugins/outputs/syslog/syslog_mapper.go

```diff
@@ -0,0 +1,193 @@
+package syslog
+
+import (
+	"errors"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/go-syslog/rfc5424"
+	"github.com/influxdata/telegraf"
+)
+
+type SyslogMapper struct {
+	DefaultSdid         string
+	DefaultSeverityCode uint8
+	DefaultFacilityCode uint8
+	DefaultAppname      string
+	Sdids               []string
+	Separator           string
+	reservedFields      map[string]bool
+}
+
+// MapMetricToSyslogMessage maps metrics tags/fields to syslog messages
+func (sm *SyslogMapper) MapMetricToSyslogMessage(metric telegraf.Metric) (*rfc5424.SyslogMessage, error) {
+	msg := &rfc5424.SyslogMessage{}
+
+	sm.mapPriority(metric, msg)
+	sm.mapStructuredData(metric, msg)
+	sm.mapAppname(metric, msg)
+	mapHostname(metric, msg)
+	mapTimestamp(metric, msg)
+	mapMsgID(metric, msg)
+	mapVersion(metric, msg)
+	mapProcID(metric, msg)
+	mapMsg(metric, msg)
+
+	if !msg.Valid() {
+		return nil, errors.New("metric could not produce valid syslog message")
+	}
+	return msg, nil
+}
+
+func (sm *SyslogMapper) mapStructuredData(metric telegraf.Metric, msg *rfc5424.SyslogMessage) {
+	for key, value := range metric.Fields() {
```

Comment:
Use `metric.FieldList()` for performance reasons, since Fields creates a new map.